avocado/core/data_dir.py

```diff
@@ -193,15 +193,23 @@ class _TmpDirTracker(Borg):
 
     def __init__(self):
         Borg.__init__(self)
+        self.keep_tmp = False
+
+    def get(self, basedir, keep_tmp):
+        # If someone sets the tmp_dir to be kept, further
+        # calls to get()  will not unset that.
+        if not self.keep_tmp:
+            self.keep_tmp = keep_tmp
 
-    def get(self):
         if not hasattr(self, 'tmp_dir'):
-            self.tmp_dir = tempfile.mkdtemp(prefix='avocado_')
+            self.tmp_dir = tempfile.mkdtemp(prefix='avocado_',
+                                            dir=basedir)
```

Comment:
In that case I'd actually only use the `basedir` option. If specified I'd keep it, if not I'd remove it afterwards. Anyway the features I'd like to see here are:

1. simple API allowing `None` as `DoNotCare`
2. in first round create the dir in basedir only when keep-tmp is specified
3: 2+ times check if params patches (or None) and warn the user when the basedir/keep-tmp changed. No checking when the user specifies `None`.

As for the setting vs. args, args should support True|False|None to enable/disable/setting and it should only try to obtain it from settings when the args says `None`. (None could be only available via defaults)